### ISSUE TITLE
Left Join & undefined fields

### ISSUE THREAD

**Issue by olintononthemovesw** [ujjwalguptaofficial/JsStore#125]
When performing a left join, with multiple records that fail the join parameters, there is an error when applying an 'as' option
Example (https://ujjwalguptaofficial.github.io/idbstudio/?db=Demo)
Initialise with:
```
update({
    in: "Orders",
    set: {
        shipperId: 1234567890
    },
    where: [{
            orderId: 10248
        },
        {
            or: {
                orderId: 10249
            }
        }
    ]
});
```
Then
```
select({
    from: "Orders",
    join: {
        with: "Shippers",
        on: "Shippers.shipperId = Orders.shipperId",
        type: "left"
    }
});
```
Successfully returns the joined records, with null field values for the aforementioned records. 
![image](https://user-images.githubusercontent.com/54812461/64409681-48aa7500-d081-11e9-92c0-34d70332bdd6.png)

But.... Applying an 'as' statement...
```
select({
    from: "Orders",
    join: {
        with: "Shippers",
        on: "Shippers.shipperId = Orders.shipperId",
        as: {
            shipperId: "Shippers.shipperId",
            shipperName: "Shippers.shipperName",
            phone: "Shippers.phone"
        },
        type: "left"
    }
});
```
![image](https://user-images.githubusercontent.com/54812461/64409706-5102b000-d081-11e9-9170-f061cab726b4.png)

**Comment by ujjwalguptaofficial** [comment]
Fixed in v - 3.3.8 . Please test and let me know.

**Comment by olintononthemovesw** [comment]
Good Work! This is working and passing our tests too. Thank you.
